CMakeLists.txt

```diff
@@ -31,11 +31,6 @@ if (CMAKE_CXX_COMPILER_ID MATCHES Clang)
         "${CMAKE_CXX_FLAGS_RELEASE} -Wno-unused-parameter")
 endif()
 
-if ((NOT "${CMAKE_CXX_FLAGS_RELEASE}" MATCHES "std=c\\+\\+0x") AND
-    (NOT "${CMAKE_CXX_FLAGS_RELEASE}" MATCHES "std=c\\+\\+11"))
-    message(FATAL_ERROR "CXX flags not set properly. Remove CMakeCache.txt and run cmake again")
-endif()
-
```

Comment:
I think this is not needed anymore, since Intel compiler also accepts `--std=c++11`. The PGI compiler however seems to accept `--c++11`, and MSVC doesn't have this option either (it supports C++11 out of the box). So we still need to rework this part.

@isuruf, you implemented this in da20e06d6caf24b63a9588069e9aa5f68ee994b6, is there any reason why we cannot just remove these lines? We still check C++11 support by compiling `cmake/checkcxx11.cpp`.